sdk/botservice/Azure.ResourceManager.BotService/src/Custom/ArmBotServiceModelFactory.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using Azure.Core;
+using Azure.ResourceManager.Models;
+using Azure.ResourceManager.Resources.Models;
+
+namespace Azure.ResourceManager.BotService.Models
+{
+    public static partial class ArmBotServiceModelFactory
+    {
+        /// <summary> Initializes a new instance of <see cref="Models.BotConnectionSettingProperties"/>. </summary>
+        /// <param name="clientId"> Client Id associated with the Connection Setting. </param>
+        /// <param name="settingId"> Setting Id set by the service for the Connection Setting. </param>
+        /// <param name="clientSecret"> Client Secret associated with the Connection Setting. </param>
+        /// <param name="scopes"> Scopes associated with the Connection Setting. </param>
+        /// <param name="serviceProviderId"> Service Provider Id associated with the Connection Setting. </param>
+        /// <param name="serviceProviderDisplayName"> Service Provider Display Name associated with the Connection Setting. </param>
+        /// <param name="parameters"> Service Provider Parameters associated with the Connection Setting. </param>
+        /// <param name="provisioningState"> Provisioning state of the resource. </param>
+        /// <returns> A new <see cref="Models.BotConnectionSettingProperties"/> instance for mocking. </returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static BotConnectionSettingProperties BotConnectionSettingProperties(string clientId = null, string settingId = null, string clientSecret = null, string scopes = null, string serviceProviderId = null, string serviceProviderDisplayName = null, IEnumerable<BotConnectionSettingParameter> parameters = null, string provisioningState = null)
+        {
+            parameters ??= new List<BotConnectionSettingParameter>();
+
+            return new BotConnectionSettingProperties(
+                id: null,
+                name: null,
+                clientId,
+                settingId,
+                clientSecret,
+                scopes,
+                serviceProviderId,
+                serviceProviderDisplayName,
+                parameters?.ToList(),
+                provisioningState,
+                serializedAdditionalRawData: null);
```

Comment:
Please change the implementation here to call the generated overload instead.

The reason of doing this change is that this method now is calling an **internal** ctor, which might change and/or have breaking changes when there is a new version.
Instead we should call a public method because we guarantee public methods will never receive breaking changes, therefore such an invocation would always work and we would never need to fix this customization code in the future.